obsreport/obsreport_scraper.go

```diff
@@ -91,10 +152,15 @@ func (s *Scraper) EndMetricsOp(
 	span := trace.SpanFromContext(scraperCtx)
 
 	if s.level != configtelemetry.LevelNone {
-		stats.Record(
-			scraperCtx,
-			obsmetrics.ScraperScrapedMetricPoints.M(int64(numScrapedMetrics)),
-			obsmetrics.ScraperErroredMetricPoints.M(int64(numErroredMetrics)))
+		if s.useOtelForMetrics {
+			s.scrapedMetricsPoints.Add(scraperCtx, int64(numScrapedMetrics), s.otelAttrs...)
+			s.erroredMetricsPoints.Add(scraperCtx, int64(numErroredMetrics), s.otelAttrs...)
+		} else { // OC for metrics
+			stats.Record(
+				scraperCtx,
+				obsmetrics.ScraperScrapedMetricPoints.M(int64(numScrapedMetrics)),
+				obsmetrics.ScraperErroredMetricPoints.M(int64(numErroredMetrics)))
+		}
```

Comment:
Should be better to move this to a `s.recordMetrics(scraperCtx, numScrapedMetrics, numErroredMetrics)` method, to encapsulate this temporary dual instrumentation and not pollute the `EndMetricsOp` method.

Reply:
Makes sense, fixed!